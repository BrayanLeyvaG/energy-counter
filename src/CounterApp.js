import React, {useState} from 'react';
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState( value ); //[]

    const handleAdd = () => setCounter(counter + 1 );

    const handleSubtract = () => setCounter(counter - 1 );

    const handleReset = () => setCounter( value );

    const handleEndTurn = () => setCounter(counter + 2 );


    

    return (
        <div className='allbody'>
            <h1>Energy Counter</h1>
            <h2> { counter } </h2>
            <div>
                <button className='b' onClick={ handleSubtract }>-1</button>
                <button className='b' onClick={ handleEndTurn }>End Turn</button>
                <button className='b' onClick={ handleAdd }>+1</button>
            </div>
                <button className='br' onClick={ handleReset }>Reset</button>
        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp 



