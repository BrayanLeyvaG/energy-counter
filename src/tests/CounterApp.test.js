import React from 'react';
import { shallow } from "enzyme"
import CounterApp from "../CounterApp";


describe('Probando <CounerApp />', () => {
    
    let wrapper
    
    beforeEach( () => {
         wrapper = shallow(<CounterApp value = { 3 } />)

    }

    )

    test('Debe mostrar <CounerApp /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar el valor por defecto de 100', () => {
        const value = 100
        const wrapper = shallow(<CounterApp value = {value} />)

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe(value.toString());
    })

    test('Debe de incrementar con el boton +1', () => {
        wrapper.find('button').at(2).simulate('click')
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('4')
    })

    test('Debe de decrementar con el boton -1', () => {
        wrapper.find('button').at(0).simulate('click')
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('2')
    })
    
    test('debe de colocar el valor por defecto con el btn reset', () => {
        const value = 3
        const wrapper = shallow(<CounterApp value = {value} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('.br').simulate('click')
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(value.toString())
    })
    
    

})
