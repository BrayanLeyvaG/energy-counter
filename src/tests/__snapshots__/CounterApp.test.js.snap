// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <CounerApp /> Debe mostrar <CounerApp /> correctamente 1`] = `
<div
  className="allbody"
>
  <h1>
    Energy Counter
  </h1>
  <h2>
     
    3
     
  </h2>
  <div>
    <button
      className="b"
      onClick={[Function]}
    >
      -1
    </button>
    <button
      className="b"
      onClick={[Function]}
    >
      End Turn
    </button>
    <button
      className="b"
      onClick={[Function]}
    >
      +1
    </button>
  </div>
  <button
    className="br"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
